modules/animation/src/animationcontroller.cpp

```diff
@@ -479,6 +483,53 @@ const AnimationState& AnimationController::getState() const { return state_; }
 
 Seconds AnimationController::getCurrentTime() const { return currentTime_; }
 
+void AnimationController::resetAllPoperties() {
+    auto options = util::transform(app_->getDataWriterFactory()->getExtensionsForType<Layer>(),
+                                   [](const auto& i) -> std::string { return toString(i); });
+
+    size_t selectedIndex = [&]() -> size_t {
+        auto ext = app_->getDataWriterFactory()->getExtensionsForType<Layer>();
+        auto it =
+            std::find_if(ext.begin(), ext.end(), [](auto& e) { return e.extension_ == "png"; });
+        if (it != ext.end())
+            return std::distance(ext.begin(), it);
+        else
+            return 0;
+    }();
+
+    renderImageExtension.replaceOptions(options);
+    renderImageExtension.setSelectedIndex(selectedIndex);
+    renderImageExtension.setCurrentStateAsDefault();
+
+    PropertyOwner::resetAllPoperties();
+}
+
+void AnimationController::serialize(Serializer& s) const {
+    PropertyOwner::serialize(s);
+    s.serialize("renderImageExtension", renderImageExtension.getSelectedIndex());
```

Comment:
Would it not make more sense to serialize the extension instead of the index?

Reply:
true